core/src/main/java/io/grpc/internal/Channelz.java

```diff
@@ -449,22 +455,125 @@ public ChannelStats build() {
     }
   }
 
+  public interface SecurityInfoProvider {
+    Security get();
+  }
+
+  public static final SecurityInfoProvider NO_SECURITY_INFO = new SecurityInfoProvider() {
+    @Override
+    public Security get() {
+      return null;
+    }
+  };
+
+  public static final class TlsSecurityInfoProvider implements SecurityInfoProvider {
+    private final SSLSession session;
+
+    public TlsSecurityInfoProvider(SSLSession session) {
+      this.session = Preconditions.checkNotNull(session, "session");
+    }
+
+    @Override
+    public Security get() {
+      return new Security(new Tls(session));
+    }
+  }
+
   public static final class Security {
-    // TODO(zpencer): fill this in
+    @Nullable
+    public final Tls tls;
+    @Nullable
+    public final OtherSecurity other;
+
+    public Security(Tls tls) {
+      this.tls = tls;
```

Comment:
Is there an intention for this to be non-null? Ditto for `other`.

Reply:
Yes, exactly one of the two should be non null. Added precondition.

Reply:
Yes, exactly one should be non null. Added precondition.